graspologic/plot/plot.py

```diff
@@ -673,6 +676,222 @@ def pairplot(
     return pairs
 
 
+def plot_ellipse(
+    data, X, j, k, means, covariances, ax, label_palette, cluster_palette, alpha
+):
+    r"""
+    plot_ellipse makes a scatter plot from the two dimensions j,k where j
+    corresponds to x-axis
+    and k corresponds to the y-axis onto the axis that is ax. plot_ellipse then
+    applies a gmm ellipse onto the scatterplot
+    using the data from Y_, means, covariances.
+
+    Parameters
+    ----------
+    X : array-like, shape (n_samples, n_features)
+        Input data.
+    j : int
+        column index of feature of interest from X which will be the x-axis data.
+    k : int
+        column index of feature of interest from X which will be the y-axis data.
+    means : array-like, shape (n_components, n_features)
+        Estimated means from gmm
+    covariances : array-like, shape (with 'full') (n_components, n_features, n_features)
+        estimated covariances from gmm
+    ax : axis object
+        The location where plot_ellipse will plot
+    label_palette : dict, optional, default: dictionary using 'Set1'
+    cluster_palette : dict, optional, default: dictionary using 'Set1'
+    alpha : float, optional, default: 0.7
+        Opacity value of plotter markers between 0 and 1
+    """
+    sns.scatterplot(
+        data=data, x=X[:, j], y=X[:, k], ax=ax, hue="labels", palette=label_palette
+    )
+    means_collector = data[[j, k, "labels"]]
+    means_collector = means_collector.groupby("labels").mean()
+    for i, (mean, covar) in enumerate(zip(means, covariances)):
+        v, w = linalg.eigh(covar)
+        v = 2.0 * np.sqrt(2.0) * np.sqrt(v)
+        u = w[0] / linalg.norm(w[0])
+        # Plot an ellipse to show the Gaussian component
+        angle = np.arctan(u[1] / u[0])
+        angle = 180.0 * angle / np.pi
+        for i, m in means_collector.iterrows():
+            if abs(float(round(mean[j], 3)) - float(round(m[j], 3))) < 0.1:
+                if abs(float(round(mean[k], 3)) - float(round(m[k], 3))) < 0.1:
+                    color = cluster_palette[i]
```

Comment:
There is an is here with closeness. this is why my tests are failing... unsure how to check for closeness of means other ways.

Reply:
still don't understand the fundamental problem this is trying to solve so not sure what to recommend

Reply:
for one separable group of the data, the labeled data has one mean and the gmm clustered data has another mean. What this aims to do is to find the labeled mean data closest to the current clustered mean data. and then retrieve the color of that labeled data and color the ellipse the same thing.

Reply:
I handle this in a better way in the push I just made.

Reply:
@dfrancisco1998 I did something extremely similar to this in one of my tests: https://github.com/loftusa/graspologic/blob/oos-ase/tests/test_spectral_embed.py#L206

Reply:
@loftusa will def look! but i no longer need to match colors like this!